### ISSUE TITLE
attaching to container fails since last update

### ISSUE THREAD

**Issue by flexo3001** [tomsquest/docker-radicale#70]
Since the last update i'm getting the following error:
![Screenshot from 2020-03-09 15-03-50](https://user-images.githubusercontent.com/1415596/76220302-8a2a0180-6217-11ea-9c55-0a2c39698b34.png)

docker-compose file which worked before:
```
version: "2.2"

services:
  server:
    container_name: radicale
    image: tomsquest/docker-radicale:latest
    restart: unless-stopped
    read_only: true
    volumes:
      - /home/foo/radicale/data:/data
      - /home/foo/radicale/config:/config:ro
    ports:
      - "127.0.0.1:5232:5232"
    security_opt:
      - no-new-privileges:true
    cap_drop:
      - ALL
    cap_add:
      - CHOWN
      - SETUID
      - SETGID
      - KILL
    mem_limit: 256M
    pids_limit: 50
    read_only: true
    init: true
```

**Comment by tomsquest** [comment]
Hi @flexo3001 ,

Can you make another try with tag `2.1.11.5` (instead of `latest`)?

Maybe #68 #69 have broken something :/

**Comment by tomsquest** [comment]
I do not seem to get any error.

With the same docker-compose as yours (untouched):

```
mkdir -p /tmp/radicale/{data,config}
cp ~/Dev/docker-radicale/config /tmp/radicale/config
$ docker-compose up
Creating radicale ... done
Attaching to radicale
# no errors
```

**Comment by flexo3001** [comment]
Hey, i tested `2.1.11.5` and it produces the same output. If i use `2.1.11.4` everything works fine. I tested on my Debian 10 + Docker repo (19.03.7)  server and on my Fedora 31 + Docker repo (19.03.7) desktop and it fails on both. :(

**Comment by SpaceMaster85** [comment]
Hello 
I have the same problem.
The container stoped working after the last update.
If I remove the` --read-only` flag the container starts, with this flag set I get the following error

```
groupmod: cannot open /etc/group
groupmod: failed to unlock /etc/group
```

I used "Option 1. User/Group 2999 on the host"

**Comment by flexo3001** [comment]
I'm using "Option 0. Do nothing, the container will fix the permission itself" and according to the folder/file permissions 2999 is/was being used. Without `read-only` the container starts.

**Comment by tomsquest** [comment]
Hi @flexo3001 and @SpaceMaster85 ,

Sorry to have broken something on your side. Obviously it was not intended.

The `groupmod` error that is printed _should only occur_ if an env var named `PGID` exists. As per `docker-entrypoint.sh`:

```

if [ -n "$PGID" ]; then
    groupmod -o -g "$PGID" radicale
fi
```
(BTW, same for the `PID` env var).

Can you check your ENV for these variables : `env | grep ID=` should print stuff, but no `PUID` nor `PGID`.

If your environment contains these variables, can you try by unsetting them (`unset PUID;unset PGID`).

If your environment does not contain these vars, I am a bit loss.

Can you test by forcing the env directly in docker-compose.yml (as below), or with `-e PUID=2999 -e PGID=2999` ?

```
environment:
  - PUID=2999
  - PGID=2999
```

**Comment by tomsquest** [comment]
I managed to reproduce. I think my `latest` was not the pushed latest. 
Investigating...

**Comment by SpaceMaster85** [comment]
You don't need to apologize to us! We have to thank you for your work and your effort! 

Thank you very much!

**Comment by flexo3001** [comment]
Yeah, what @SpaceMaster85 said. Thank you for the container and for looking into it!

**Comment by adzero** [comment]
@tomsquest 

Sorry to have triggered this issue with my (uncomplete) PR changes. 

Let me know if I can help anyway.

**Comment by tomsquest** [comment]
Hi everyone,

I created PR #71 . 

The best fix I was able to make is to split build and run args by name. 
So now, build args are named (obviously) `BUILD_UID` and `BUILD_GID`.
Run args are back to their defaults: `UID` and `GID`.

@adzero when this PR will be merged and the next release pushed, you will have to change your settings on Synology. (PUID -> BUILD_PID and PGID -> BUILD_GID). 
@adzero can you test with this branch ? If this is hard/impossible on your side, I will merge it as is.

**Comment by adzero** [comment]
Hi @tomsquest,

Sorry for the delay, my mind was somewhere else the last few days. Anyway, I finally found time to refocus on this. 

I pulled your branch in my repo and made a new build using the new BUILD_UID/BUILD_GID parameters. Build process works like a charm. 
I tried to run the container and did  a quick Radicale setup to check ownership put on the created files. The user and group are the ones that I passed at build time. 
So I suppose everything is OK on my side. 

Once changes are merged into your repo's master branch I will create a clean production setup to migrate my contacts and calendar in Radicale. 

Thanks again for your work.

**Comment by tomsquest** [comment]
Hi @adzero , Thanks for the feedback.

I will try to add some more automated tests before merging the PR (production mode tests certainly).

**Comment by stackcoder** [comment]
Until the fixes are released to docker hub, its possible to run the image using following workaround:
```yml
environment:
- PUID=
- PGID=
read_only: true
cap_drop:
  - ALL
cap_add:
  - SETUID
  - SETGID
  - CHOWN
  - DAC_OVERRIDE
  - KILL
```

@tomsquest Should the `DAC_OVERRIDE` capability be required and added to the docker-compose example? I was unable to run the image without it.

**Comment by tomsquest** [comment]
@stackcoder using `DAC_OVERRIDE` is nearly as bypassing the cap_drop, so not recommended for a "production" run.

> CAP_DAC_OVERRIDE : Bypass file read, write, and execute permission checks.  
> (DAC is an abbreviation of "discretionary access control".)

**Comment by tomsquest** [comment]
Released in **2.1.11.6**.

Sorry for the time it took. Wanted to add a test with production instruction, broke my python setup, restarted from scratch, and ultimately still failed.
